### PR TITLE
add compatibility scraper for karpenter and util `validate_semver`

### PR DESCRIPTION
**Pull request by seemywingz**
add compatibility scraper for karpenter and util `validate_semver`

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.

### CODE REVIEW

File: utils/compatibility/utils.py [hunk 1]
```diff
@@ -52,6 +72,21 @@ def write_yaml(file_path, data):
...
 
 
+def validate_semver(version_str):
+    try:
+        # Coerce the version string to handle versions like "1.30", "1.27", or "1.2.3"
+        version = semantic_version.Version.coerce(version_str)
+
+        # Check if prerelease and build are empty
+        if version.prerelease or version.build:
+            return False
```
Comment (michaeljguarino) [inline utils/compatibility/utils.py:82]: `return None` would probably be a slightly more natural typing for this function (shouldn't affect the behavior any though)

**Review by michaeljguarino — COMMENTED** [review]
mostly lgtm, minor fix though

File: utils/compatibility/utils.py (hunk 1 above)
Comment (seemywingz) [reply utils/compatibility/utils.py]: Ack, that makes sense.

**Comment by michaeljguarino** [comment]
The test that's failing specifies a specific kube version, and just needs to be rewritten now that it's moved to 1.31.  Make that change and this is good to merge.

File: test/console/graphql/queries/deployments/cluster_queries_test.exs [hunk 2]
```diff
@@ -583,6 +583,7 @@ defmodule Console.GraphQl.Deployments.ClusterQueriesTest do
       admin = admin_user()
       insert_list(2, :cluster, current_version: "1.29", pinged_at: Timex.now(), upgrade_plan: %{compatibilities: true, incompatibilities: true, deprecations: true})
       insert_list(3, :cluster, current_version: "1.30", pinged_at: Timex.now() |> Timex.shift(days: -1))
+      insert_list(2, :cluster, current_version: "1.31", pinged_at: Timex.now() |> Timex.shift(days: -1))
```
Comment (michaeljguarino) [inline test/console/graphql/queries/deployments/cluster_queries_test.exs:586]: the reason you had to change the asserts is you didn't modify the top set of clusters to `1.30` and the following to `1.31` to preserve the version pattern

Comment (seemywingz) [reply test/console/graphql/queries/deployments/cluster_queries_test.exs]: this looks better: https://github.com/pluralsh/console/pull/1524/files#diff-9356619c481fa6b7722ae779fe4a8d30d3038acfe35db254f7db13039f6e0cee

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**